src/mono/mono/metadata/sre.c

```diff
@@ -49,6 +49,10 @@
 #endif
 #include "icall-decl.h"
 
+/* Maps MonoMethod* to weak links to DynamicMethod objects */
+/* Protected by the domain lock */
+static GHashTable *method_to_dyn_method;
```

Comment:
I wonder if this should go into ALCs. Dynamic methods have some association to assemblies that they can access, and presumably when that's unloaded, they need to be cleaned up.  Is that right @CoffeeFlux ?

Reply:
Dynamic methods are handled specially, they are kinda outside alcs, we clean up all their memory manually when they are deleted.